proxy-node-shared/src/test/java/uk/gov/ida/notification/shared/proxy/VSPProxyTranslateResponseTest.java

```diff
@@ -0,0 +1,149 @@
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.UriBuilder;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+import static org.mockito.ArgumentMatchers.eq;
+
+@RunWith(MockitoJUnitRunner.class)
+public class VSPProxyTranslateResponseTest {
+
+    @Path("/")
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.APPLICATION_JSON)
+    public static class TestVSPResource {
+
+        @POST
+        @Path("/translate-response")
+        public TranslatedHubResponse testGetTranslatedHubResponse(VerifyServiceProviderTranslationRequest request) {
+            if ("SAMLResponse1234".equals(request.getSamlResponse())) {
+                return TranslatedHubResponseBuilder.getTranslatedHubResponseIdentityVerified();
+            } else {
+                return TranslatedHubResponseBuilder.getTranslatedHubResponseAuthenticationFailed();
+            }
+        }
+    }
+
+    @Path("/translate-response")
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.APPLICATION_JSON)
+    public static class TestVSPServerErrorResource {
+
+        @POST
+        public Response testGenerate(AuthnRequestGenerationBody authnRequestGenerationBody) {
+            return Response.serverError().build();
+        }
+    }
+
+    @Path("/translate-response")
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.APPLICATION_JSON)
+    public static class TestVSPClientErrorResource {
+
+        @POST
+        public Response testGenerate(AuthnRequestGenerationBody authnRequestGenerationBody) {
+            return Response.status(Response.Status.BAD_REQUEST).build();
+        }
+    }
+
+    @ClassRule
+    public static final DropwizardClientRule testVspClientRule = new DropwizardClientRule(new TestVSPResource());
+
+    @ClassRule
+    public static final DropwizardClientRule testVspServerErrorClientRule = new DropwizardClientRule(new TestVSPServerErrorResource());
+
+    @ClassRule
+    public static final DropwizardClientRule testVspClientErrorClientRule = new DropwizardClientRule(new TestVSPClientErrorResource());
+
+    @Spy
+    JsonClient jsonClient = new JsonClient(
+            new ErrorHandlingClient(ClientBuilder.newClient()),
+            new JsonResponseProcessor(new ObjectMapper())
+    );
+
+    @Test
+    public void shouldReturnTranslatedHubResponseAuthnFailed() {
+
+        VerifyServiceProviderProxy vspProxy = new VerifyServiceProviderProxy(jsonClient, testVspClientRule.baseUri());
+
+        TranslatedHubResponse response =
+                vspProxy.getTranslatedHubResponse(
+                        new VerifyServiceProviderTranslationRequest(null, "_1234", "LEVEL_2")
+                );
+
+        assertEquals(VspScenario.AUTHENTICATION_FAILED, response.getScenario());
+        assertEquals("pid1234", response.getPid());
+        assertEquals(VspLevelOfAssurance.LEVEL_2, response.getLevelOfAssurance());
+        assertEquals(null, response.getAttributes());
```

Comment:
Can use `assertNull` to have a better error message

Reply:
Done.